### DISCUSSION TITLE
How to set a bolean value?

### DISCUSSION THREAD

**Discussion by TomasTrnkaPLC** [FreeOpcUa/python-opcua#1193] [Q&A]
Hi I try to set a value like this

` PC_Valve_1_1_Close_var = root.get_child(["0:Objects", "4:new_Controller_0", "3:GlobalVars","4:PC_Valve_1_1_Close"])
        PC_Valve_1_1_Close_var = PC_Valve_1_1_Close_var.get_value()
        print (PC_Valve_1_1_Close_var)
        PC_Valve_1_1_Close_var.set_value(False)`
But I get an error. AttributeError: 'bool' object has no attribute 'set_value'
Why?

**Comment by starturtle** [comment]
Because you overwrite the variable with is value. Store the result of `get_value()` to something else, and you should be good again...

**Reply by AndreasHeine** [reply]
line1: "PC_Valve_1_1_Close_var" is from type "NodeClass"
line2: "PC_Valve_1_1_Close_var" is from type "Boolean"
line3: "PC_Valve_1_1_Close_var" of type Boolean has no methods related to opc ua

**Comment by zerox1212** [comment]
I tell this to everyone that is new. Use a high quality IDE like PyCharm so you can set break points and inspect the objects. OPCUA has many complex objects that are not easily understood without inspection.

**Reply by AndreasHeine** [reply]
quality IDE? PyCharm? ;) just kidding!

**Reply by swamper123** [reply]
👴 Well in my days, we hadn't IDEs! We punched everything in Cards and hoped, that we could run it with a pencil and a record player. :^)

**Reply by zerox1212** [reply]
> quality IDE? PyCharm? ;) just kidding!

JetBrains have the best IDE suite by far.  Would love to know what is better lol.

**Comment by TomasTrnkaPLC** [comment]
Many thanks now are it ok. I have one more question how to wrote a string?
By the way, I use a VisualStudio Code because I need sometimes skip between languages and is a bit difficult to keep up-to-date a many IDE.
However thanks

**Reply by AndreasHeine** [reply]
```python
mystring = "THIS IS A STRING!"
dv = ua.DataValue(ua.Variant(mystring, ua.VariantType.String))
var.set_value(dv)
```

FYI:
if you work with plc's you should always provide the propper variable type
DataValue: Value + Timestamp
Variant: Defines if its a single value or an array or multidimensional array
VariantType: is the datatype of the opc ua variable
